File: Schriftlich/Fachtheorie/SEW/SEW_3AHIT/School Projects/Party/src/Party/Person.java
package Party;

public class Person
{
    private String firstname;
    private String lastname;

    public String name;
    public Company worksAt;
    public int earnings;

    Person(Company company, int earnings)
    {
        this.worksAt = company;
        this.earnings = earnings;
        setFirstname("");
        setLastname("");
    }

    Person(String firstname, String lastname, Company company, int earnings)
    {
        this.worksAt = company;
        this.earnings = earnings;
        setFirstname(firstname);
        setLastname(lastname);
    }

    public void setFirstname(String firstname)
    {
        this.firstname = firstname;
    }

    public void setLastname(String lastname)
    {
        this.lastname = lastname;
    }

    public String getFirstname()
    {
        return firstname;
    }

    @Override
    public String toString()
    {
        return this.firstname + " " + this.lastname;
    }
}

File: Schriftlich/Fachtheorie/SEW/c# projekte/LinqQueries/LinqQueries/LinqQueries/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LinqQueries
{
    class Person
    {
        public int Nr {  get; set; }
        public string Firstname { get; set; } = string.Empty;
        public string Lastname { get; set; } = string.Empty;
        public int Group {  get; set; }
        public int Age { get; set; }
        public List<int> Grading { get; set; } = new List<int>();

        public Person(int nr, string firstname, string lastname, int group, int age, List<int> grading)
        {
            Nr = nr;
            Firstname = firstname;
            Lastname = lastname;
            Group = group;
            Age = age;
            Grading = grading ?? new List<int>(); 
        }

        public override string ToString()
        {
            return $"Nr: {Nr}, Name: {Firstname} {Lastname}, Group: {Group}, Age: {Age}, Grades: [{string.Join(", ", Grading)}]";
        }

        public static void PrintPeople(List<Person> people)
        {
            if (people.Count == 0)
            {
                Console.WriteLine("Empty");
                return;
            }

            Console.WriteLine("List of People:");
            Console.WriteLine("-------------------------------------------------------------");
            foreach (var person in people)
            {
                Console.WriteLine(person.ToString());
            }
            Console.WriteLine("-------------------------------------------------------------");
        }

    }
}
